File: SSUMobile/Modules/About/SSUAboutModule.m
//
//  SSUAboutModule.m
//  SSUMobile
//
//  Created by Eric Amorde on 9/9/15.
//  Copyright (c) 2015 Sonoma State University Department of Computer Science. All rights reserved.
//

#import "SSUAboutModule.h"

@implementation SSUAboutModule

+ (instancetype) sharedInstance {
    static SSUAboutModule * instance = nil;
    static dispatch_once_t token;
    dispatch_once(&token, ^{
        instance = [[self alloc] init];
    });
    return instance;
}

#pragma mark - SSUModule

- (nonnull NSString *) title {
    return NSLocalizedString(@"About",
                             @"General information about the app. \'Info\'");
}

- (nonnull NSString *) identifier {
    return @"about";
}

- (UIView *) viewForHomeScreen {
    UIButton * button = [UIButton buttonWithType:UIButtonTypeInfoLight];
    return button;
}

- (UIImage *) imageForHomeScreen {
    return nil;
}

- (BOOL) showModuleInNavigationBar {
    return YES;
}

- (BOOL) shouldNavigateToModule {
    return YES;
}

- (UIViewController *) initialViewController {
    UIStoryboard * storyboard = [UIStoryboard storyboardWithName:@"About"
                                                          bundle:[NSBundle bundleForClass:[self class]]];
    return [storyboard instantiateInitialViewController];
}

@end

File: SSUMobile/Modules/About/SSUAboutModule.swift
//
//  SSUAboutModule.swift
//  SSUMobile
//
//  Created by Eric Amorde on 3/25/17.
//  Copyright © 2017 Sonoma State University Department of Computer Science. All rights reserved.
//

import Foundation

class SSUAboutModule: SSUModuleBase, SSUModuleUI {
    
    @objc(sharedInstance)
    static let instance = SSUAboutModule()
    
    private let feedbackSubmissionInterval: TimeInterval = 60
    
    /**
     True if the required amount of time between feedback submissions has passed
     */
    var canSubmitFeedback: Bool {
        get {
            let lastSubmission = SSUConfiguration.sharedInstance().lastFeedbackDate
            let timeSinceLast = abs(lastSubmission.timeIntervalSinceNow)
            return timeSinceLast >= feedbackSubmissionInterval
        }
    }
    
    // MARK: SSUModule
    
    var title: String {
        return NSLocalizedString("About", comment: "General information about the app.")
    }
    
    var identifier: String {
        return "about"
    }
    
    func setup() {
        
    }
    
    // MARK: SSUModuleUI
    
    func imageForHomeScreen() -> UIImage? {
        return nil
    }
    
    func viewForHomeScreen() -> UIView? {
        let button = UIButton(type: .infoLight)
        return button
    }
    
    func initialViewController() -> UIViewController {
        let storyboard = UIStoryboard(name: "About", bundle: Bundle(for: type(of: self)))
        return storyboard.instantiateInitialViewController()!
    }

    func shouldNavigateToModule() -> Bool {
        return true
    }
    
    func showModuleInNavigationBar() -> Bool {
        return true
    }
}
